pwnlib/tubes/tube.py

```diff
@@ -632,10 +632,10 @@ def recvline_endswith(self, delims, keepends=False, timeout=default):
             b'Kaboodle'
         """
         # Convert string into singleton tupple
-        if isinstance(delims, (bytes, six.text_type)):
+        if isinstance(delims, (bytes, bytearray, six.text_type)):
             delims = (delims,)
 
-        delims = tuple(context._encode(delim) + self.newline for delim in delims)
+        delims = tuple(context._need_bytes(delim) + self.newline for delim in delims)
```

Comment:
Would really prefer to have a `pwnlib.bytes.as_bytes` instead of `context._xxx`

Reply:
I am moving them to pwnlib.util.packing, since they are kind of packing-related

Reply:
That sounds ideal.  It would be nice if the "give me bytes" function could be rolled into `flat`, since that's a paradigm used elsewhere, and has a better expected behavior than `_encode` / `_decode` and is shorter than `_need_bytes`.